### PR TITLE
chore: Refactored code location for iterators

### PR DESCRIPTION
**Pull request by jonaslagoni**
**Description**
This PR moves iterators and constants to move towards a model which only contains the model code and not excess functionality to reduce complexity. This is a pre-step for me to move assignment of anonymous ids to schemas in #223

### CODE REVIEW

**Comment by jonaslagoni** [comment]
Let me know if we should move the files into another directory or if its fine where it is

**Comment by jonaslagoni** [comment]
Also, do we need new unit tests for these functions? 🤔

File: lib/constants.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+const xParserMessageName = 'x-parser-message-name';
+const xParserSchemaId = 'x-parser-schema-id';
+const xParserCircle = 'x-parser-circular';
+const xParserCircleProps = 'x-parser-circular-props';
+
+module.exports.xParserMessageName = xParserMessageName;
```
Comment (magicmatatjahu) [inline lib/constants.js:6]: Maybe we simplify this?

```js
module.exports = {
  xParserMessageName: 'x-parser-message-name',
  ...
}
```

File: lib/iterators.js [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+      schema.items().forEach(arraySchema => {
+        traverseSchema(arraySchema, callback, null, schemaTypesToIterate);
+      });
+    } else {
+      traverseSchema(schema.items(), callback, null, schemaTypesToIterate);
+    }
+  }
+}
+
+module.exports.SchemaIteratorCallbackType = SchemaIteratorCallbackType;
```
Comment (magicmatatjahu) [inline lib/iterators.js:221]: Like in above:

```ts
module.exports = {
  SchemaIteratorCallbackType ,
  SchemaTypesToIterate ,
  ...
}
```

File: lib/constants.js (hunk 1 above)
Comment (jonaslagoni) [reply lib/constants.js]: Yea thats better 😄

**Review by magicmatatjahu — APPROVED** [review]

**Pull request merged**